rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll

```diff
@@ -484,7 +484,6 @@ module Impl {
   class VariableReadAccess extends VariableAccess {
     VariableReadAccess() {
       not this instanceof VariableWriteAccess and
-      not this = any(RefExpr re).getExpr() and
```

Comment:
I think it may be better to only consider these reads for the SSA library. Should be enough to change `certain = false` to `certain = true` [here](https://github.com/github/codeql/blob/132dbd75176b8b7a48d20475aba06193a8595085/rust/ql/lib/codeql/rust/dataflow/internal/SsaImpl.qll#L104).

Reply:
Done. I had to also handle `RefExpr` in `variableReadActual`.